docs/usage/10-provisioning-services.adoc

```diff
@@ -29,78 +29,114 @@ In particular, exposing UDP or SCTP services is **NOT** supported.
 
 To expose a TCP application running on MKE to either inside or outside the DC/OS cluster, a Kubernetes https://kubernetes.io/docs/concepts/services-networking/service/[`Service`] resource of type https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer[`LoadBalancer`] must be created.
-`dklb` will react to this event by provisioning an EdgeLB pool (henceforth referred to as the _target EdgeLB pool_) for the `Service` resource based on its specification.
+`dklb` will react to this event by provisioning an EdgeLB pool (henceforth referred to as the _target EdgeLB pool_) for the `Service` resource.
+This EdgeLB pool is provisioned using sane default values for its name, its CPU, memory and size requirements, and other options.
 
 After provisioning said EdgeLB pool, `dklb` will periodically query EdgeLB in order to obtain the list of hostnames and IPs at which the service can be reached.
 These will eventually be reported on the `.status` field of the `Service` resource.
 It should be noted that, due to the way EdgeLB pool scheduling and metadata reporting works, it may take from a few seconds to several minutes for these hostnames and IPs to be reported.
 
-=== Customizing the name of the EdgeLB pool
+=== Customizing the target EdgeLB pool
+
+As mentioned before, `dklb` uses sane defaults when provisioning EdgeLB pools for `Service` resources of type `LoadBalancer`.
+It is, however, possible to customize the target EdgeLB pool for a given `Service` resource by using the `kubernetes.dcos.io/dklb-config` annotation.
+This annotation accepts a YAML object (henceforth called the _configuration object_) with the following structure:
 
-By default, `dklb` uses the MKE cluster's name and the `Service` resource's namespace and name to compute the name of the target EdgeLB pool.
-To specify a custom name for said EdgeLB pool, one may https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/[annotate] the `Service` resource with the following annotation:
+[source,yaml]
+----
+kubernetes.dcos.io/dklb-config: |  # NOTE: The "|" character is mandatory.
+  name: dklb-pool-0
+  role: "*"
+  network: "dcos"
+  size: 3
+  cpus: 0.2
+  memory: 512
+  frontends:
+  - port: 18080
+    servicePort: 8080
+  - port: 19090
+    servicePort: 9090
+----
+
+The subsections below provide more insight on each of the fields on the configuration object.
+
+[NOTE]
+====
+The `kubernetes.dcos.io/dklb-config` is **OPTIONAL**.
+Whenever said annotation is absent, `dklb` will generate a sane default configuration for the target EdgeLB pool and automatically inject it on the `Service` resource.
+====
+
+WARNING: The `kubernetes.dcos.io/dklb-config` cannot be removed after the `Service` resource is created.
+
+==== Customizing the name of the EdgeLB pool
+
+By default, `dklb` uses the MKE cluster's name and a randomly-generated five-character suffix to compute the name of the target EdgeLB pool.
```

Comment:
Does `dklb` make sure there are no collisions?

Reply:
It doesn't, but I'd say the chances of a collision are rather low given the alphabet used to generate the suffix (i.e. `a-z0-9]`) _and_ the fact that we use the current time as the source of randomness.

Reply:
Since we can automatically generate a name for this, I think we must ensure uniqueness. Otherwise, we may run into trouble.

Reply:
I've opened #24 in order to address this concern.